### PR TITLE
feat(config): add the ability to customize commit messages

### LINKED ISSUE #153: Allow customization of commit messages & pull requests
Hi,

It would be nice to have a way to customize a few things:
- [ ] commit messages / pull requests names (through templating maybe)
- [x] pull requests labels (I use a `dependency` label for all that stuff)

### PR DESCRIPTION
**Pull request by Arcanemagus**
Add configuration options allowing Greenkeeper's commit message format to be customized as the user desires. These are handled exactly like other configuration options, as a `commitMessages` key in `package.json`. Some messages support data coming in, the format for this follows JavaScript's template literals.

This will allow Greenkeeper to fit in with any custom commit message format that a project requires, instead of forcing the conventional-commit style.

Within `package.json` the following keys are used as the default values:
```json
{
  "greenkeeper": {
    "commitMessages": {
      "initialBadge": "docs(readme): add Greenkeeper badge",
      "initialDependencies": "chore(package): update dependencies",
      "initialBranches": "chore(travis): whitelist greenkeeper branches",
      "dependencyUpdate": "fix(package): update ${dependency} to version ${version}",
      "devDependencyUpdate": "chore(package): update ${dependency} to version ${version}",
      "dependencyPin": "fix: pin ${dependency} to ${oldVersion}",
      "devDependencyPin": "chore: pin ${dependency} to ${oldVersion}",
      "closes": "\n\nCloses #${number}"
    }
  }
}
```
_Note: `closes` is somewhat special in that it is conditionally appended to (dev)`dependencyUpdate` and doesn't get used on its own._

The values within `${}` for each message are passed in from the appropriate places within the code, they can be omitted, but the variable names can't be changed from the configuration.

* `dependencyUpdate` and `devDependencyUpdate` support these parameters:
  * `dependency`
  * `version`
* `dependencyPin` and `devDependencyPin` support these parameters:
  * `dependency`
  * `oldVersion`
* `closes` supports a `number` parameter denoting the PR it closes.

As an example usage to have Greenkeeper follow ESLint convention (#608), the following configuration could be used:
```json
{
  "greenkeeper": {
    "commitMessages": {
      "initialBadge": "Docs: Add Greenkeeper badge",
      "initialDependencies": "Upgrade: Update dependencies",
      "initialBranches": "Build: Whitelist greenkeeper branches",
      "dependencyUpdate": "Upgrade: Update ${dependency} to version ${version}",
      "devDependencyUpdate": "Upgrade: Update ${dependency} to version ${version}",
      "dependencyPin": "Fix: Pin ${dependency} to ${oldVersion}",
      "devDependencyPin": "Fix: Pin ${dependency} to ${oldVersion}"
    }
  }
}
```

_Note: Since `closes` is the same as the default, there is no need to specify it here._
_Note: #608 was talking about an "(extended) ESLint" configuration, the above follows [this guide](https://eslint.org/docs/developer-guide/contributing/pull-requests#step-2-make-your-changes)._

Fixes #153.

### CODE REVIEW

**Comment by Arcanemagus** [comment]
Hopefully this works, the test scripts are completely broken on Windows 😛.

**Comment by Arcanemagus** [comment]
It looks like third party PR's don't get a valid `PRIVATE_KEY`, so they can't run.

**Comment by Realtin** [comment]
@Arcanemagus I updated the Travis config and it now runs the pr correctly. I had to 'unsecure' the PRIVATE_KEY, which is ok since it is just [a randomly generated one](https://github.com/greenkeeperio/greenkeeper/blob/master/CONTRIBUTING.md#tests) anyway!

I only had a brief look on the PR and noticed that lodash has a function `_.template` which seems to be the same as `es6-template-strings` and we already have lodash so you could use that instead. The other module seems to not be tested on newer node versions (7+).

Tests are failing unfortunately, I can try to dig into it on the weekend.

**Comment by Arcanemagus** [comment]
@Realtin Changed this over to `_.template` and fixed the code 😉, thanks for fixing the builds!

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "catbox-memory": "^2.0.4",
     "couchdb-bootstrap": "^1.14.0",
     "envalid": "^4.0.1",
+    "es6-template-strings": "^2.0.1",
```
Comment (Realtin) [inline package.json]: You missed removing the package from the package.json :)

**Comment by Realtin** [comment]
I looked at an outdated diff 😅

**Comment by jy95** [comment]
Does the bot `greenkeeper[bot] ` has the latest version ?
I set up a new [repo](https://github.com/jy95/mediaScan/blob/master/package.json#L59) with @Arcanemagus ESlint example and it seems Angular convention is still prefered.

**Comment by Realtin** [comment]
Hey,  the feature is not yet deployed to production. I'll ping you on the issue if it is.

**Comment by Realtin** [comment]
@jy95 @Arcanemagus It's released on production now! 🚀 ✨ 📝 
We will officially advertise it as soon as we wrote up some documentation for the FAQ :)

**Comment by jy95** [comment]
Thanks @Realtin Work as expected on my test repo : https://github.com/jy95/mediaScan

**Pull request merged**